src/instructlab/config/init.py

```diff
@@ -349,9 +361,12 @@ def get_gpu_or_cpu() -> tuple[str, int, int]:
         click.echo("Detecting hardware...")
         gpus = torch.cuda.device_count()
         for i in range(gpus):
-            properties = torch.cuda.get_device_properties(i)
-            chip_name = properties.name.lower()
-            total_vram += properties.total_memory  # memory in B
+            chip_name, vram = get_cuda_device_properties(i)
+            # if the SKU is something like A100-SXM4-40GB, split it for easier reading later
+            if "-" in chip_name:
+                chip_split = chip_name.split("-")
+                chip_name = " ".join(chip_split)
```

Comment:
Would be better to contain all this processing within the helper function IMO 
and use regex to split on a set of predefined separators 
not blocking

Reply:
I did it this way for testing purposes so I could mock the return value properly, I'd prefer to keep it like this if possible 🙏

Reply:
I mean sure
but IMO it would make more sense to have the unit test be for the helper function and mock it there instead